### PR TITLE
Add extension to avoid costly computation.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    internal static class SyntaxTriviaListExtensions
+    {
+       public static SyntaxTrivia LastOrDefault(this SyntaxTriviaList triviaList)
+            => triviaList.Count == 0 ? default(SyntaxTrivia) : triviaList.Last();
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs]: SyntaxTriviaList defines .Last, so this is nicely O(1) with no allocs.

Comment (sharwell) [inline src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs]: :bulb: `default`

:bulb: Maybe move the `Any` extension from `Microsoft.CodeAnalysis.CSharp.Extensions.SyntaxTriviaListExtensions` to here?

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-infrastructure .  This addresses a build break currently on jenkins.  Build break appears to be the result of merges that were independently ok, but combined cause a warning that ends up breaking hte build.

File: src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs]: SyntaxKind is C# only (though we could consider a RawKind common helper).

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs]: Moved to use default.

File: src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    internal static class SyntaxTriviaListExtensions
+    {
+       public static SyntaxTrivia LastOrDefault(this SyntaxTriviaList triviaList)
+            => triviaList.Count == 0 ? default : triviaList.Last();
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs]: :bulb: `Any()` is slightly optimized for this type compared to `Count==0`.

**Comment by CyrusNajmabadi** [comment]
I'm leaving to go to dinner.  If i'm not around to merge this in, feel free to do so.

File: src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTriviaListExtensions.cs]: good point.

**Pull request merged**